gradio/components.py

```diff
@@ -4856,6 +4856,10 @@ def postprocess(self, y) -> Dict[str, str] | None:
         if y is None:
             return None
         if isinstance(y, (ModuleType, matplotlib.figure.Figure)):  # type: ignore
+            if not matplotlib.get_backend().lower() == "agg":
+                warnings.warn(
+                    "Matplotlib backend is not set to 'Agg'. Plot may not display correctly. To fix this, add the following to the top of your script: import matplotlib; matplotlib.use('Agg')"
```

Comment:
I wonder if we can raise this somewhere up the chain (plot component init?). May be some false warnings but I worry the app would have already crashed before this is printed.

Reply:
Good point, let me think about this some more as we don't want too many false warnings.

Reply:
What if we were to try/catch the function execution, specifically looking for "main thread is not in main loop"

Reply:
Yea I think that will work